staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -91,6 +92,10 @@ func newGenericWebhook(scheme *runtime.Scheme, codecFactory serializer.CodecFact
 	codec := codecFactory.LegacyCodec(groupVersions...)
 	clientConfig.ContentConfig.NegotiatedSerializer = serializer.NegotiatedSerializerWrapper(runtime.SerializerInfo{Serializer: codec})
 
+	if customDial != nil {
```

Comment:
Is this if condition redundant?

Reply:
If `customDial` is nil, we should not reset `clientConfig.Dial`. I think we still need the if statement.

Reply:
If the pre-existing customDial matters, then logically it would need to be an argument in the construction of the new custom dial?

So this if statement scares me a bit. What breaks if it is removed?

Reply:
Well we introduced the `customDial` so egressselector is the only one using it. As discussed previously, we will be removing the `customDial` in favor of a function that uses the existing dialer as an argument to creating the new dialer. However, that will require a complete refactor of all existing egress endpoints, and I'd like to separate out the PRs so that I we don't both cover a new egress endpoint and do a refactor in the same PR.

The WIP of the new PR is here: https://github.com/kubernetes/kubernetes/pull/88539